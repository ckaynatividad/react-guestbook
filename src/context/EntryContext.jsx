import { createContext, useContext, useState } from 'react';

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([]);

  return (
    <EntryContext.Provider value={{ entries, setEntries }}>
      {children}
    </EntryContext.Provider>
  );
};

const useEntries = () => {
  const context = useContext(EntryContext);

  if (context === undefined) {
    throw new Error('must be defined with entryprovider');
  }
  return context;
};

export { EntryProvider, useEntries };
