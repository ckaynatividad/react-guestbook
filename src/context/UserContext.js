import { createContext, useContext, useMemo, useState } from 'react';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  console.log(process.env.AUTH_EMAIL);
  const login = (email, password) => {
    const loginSuccessful =
      email === process.env.AUTH_EMAIL &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccessful) setUser({ email });
    return loginSuccessful;
  };

  const logout = (callback) => {
    setUser(null);
    callback();
  };

  const value = { user, setUser, login, logout };

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};
const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('must be used within userprovider');
  }
  return context;
};

export { UserProvider, useUser };
