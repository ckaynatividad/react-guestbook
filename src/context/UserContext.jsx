import { createContext, useContext, useMemo, useState } from 'react';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState('');

  const value = useMemo(() => ({ user, setUser }), [user]);

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};
const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('must be used within userprovider');
  }
  return context;
};

export { UserProvider, useUser };
