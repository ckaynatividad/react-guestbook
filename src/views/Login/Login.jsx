import React, { useState } from 'react';
import { useLocation } from 'react-router-dom';
import { useHistory } from 'react-router-dom';
import { useForm } from '../../components/PrivateRoute';
import { useUser } from '../../context/UserContext';

export default function Login() {
  const history = useHistory();
  const location = useLocation();
  const auth = useUser();
  const { formState, handleFormChange } = useForm({ email: '', password: '' });
  const { from } = location.state || { from: { pathname: '/guestbook' } };
  const [error, setError] = useState(null);

  const handleLogin = (e) => {
    e.preventDefault();
    const loginWasSuccessful = auth.login(formState.email, formState.password);

    if (loginWasSuccessful) {
      history.replace(from.pathname);
    } else {
      setError('unsuccessful sign in');
    }
  };
  return (
    <div>
      <h1>login before seeing {from.pathname}</h1>
      <form onSubmit={handleLogin}>
        <label>
          Email
          <input
            id="email"
            name="email"
            type="email"
            onChange={handleFormChange}
          />
        </label>{' '}
        <label>
          Password
          <input
            id="password"
            name="password"
            type="password"
            onChange={handleFormChange}
          />
        </label>
        <button type="submit" onClick={handleLogin}>
          Sign In
        </button>
      </form>
      {error && <h1>{error}</h1>}
    </div>
  );
}
