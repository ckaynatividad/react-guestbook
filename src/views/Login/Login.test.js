import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { UserProvider } from '../../context/UserContext';
import Login from './Login';

test('login render', () => {
  const container = render(
    <MemoryRouter>
      <UserProvider>
        <Login />
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});

test('can login', () => {
  const container = render(
    <MemoryRouter>
      <UserProvider>
        <Login email="test@test.com" password="password" />
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
