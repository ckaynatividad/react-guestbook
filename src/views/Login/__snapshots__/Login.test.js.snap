// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can login 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          login before seeing 
          /
        </h1>
        <form>
          <label>
            Email
            <input
              id="email"
              name="email"
              type="email"
            />
          </label>
           
          <label>
            Password
            <input
              id="password"
              name="password"
              type="password"
            />
          </label>
          <button
            type="submit"
          >
            Sign In
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        login before seeing 
        /
      </h1>
      <form>
        <label>
          Email
          <input
            id="email"
            name="email"
            type="email"
          />
        </label>
         
        <label>
          Password
          <input
            id="password"
            name="password"
            type="password"
          />
        </label>
        <button
          type="submit"
        >
          Sign In
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`login render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          login before seeing 
          /
        </h1>
        <form>
          <label>
            Email
            <input
              id="email"
              name="email"
              type="email"
            />
          </label>
           
          <label>
            Password
            <input
              id="password"
              name="password"
              type="password"
            />
          </label>
          <button
            type="submit"
          >
            Sign In
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        login before seeing 
        /
      </h1>
      <form>
        <label>
          Email
          <input
            id="email"
            name="email"
            type="email"
          />
        </label>
         
        <label>
          Password
          <input
            id="password"
            name="password"
            type="password"
          />
        </label>
        <button
          type="submit"
        >
          Sign In
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
